pennylane/pauli/pauli_arithmetic.py

```diff
@@ -796,6 +800,11 @@ def __repr__(self):
         """Terminal representation for PauliSentence"""
         return str(self)
 
+    @property
+    def pauli_rep(self):
```

Comment:
I don't like this property very much. I'm assuming this is added so that we can use `qml.pauli.pauli_sentence` with any op given to `qml.dot()`. I think it's more sensible to update `pauli_sentence` directly to handle `PauliWord` and `PauliSentence` using instance checks.

Reply:
The motivation here is to do the check `if op.pauli_rep:` though technically this is not essential to this PR actually.

Reply:
Removed this property since its not actually part of this feature. I think it makes sense to add anyway at another point though.